mungegithub/mungers/submit-queue.go

```diff
@@ -800,6 +800,18 @@ const (
 	unmergeableMilestone    = "Milestone is for a future release and cannot be merged"
 )
 
+func getEarliestApprovedTime(lgtmTime, approvedTime *time.Time) *time.Time {
+	if lgtmTime == nil {
+		return approvedTime
+	} else if approvedTime == nil {
+		return lgtmTime
+	} else if lgtmTime < approvedTime {
```

Comment:
We return nil if both `lgtmTime` and `approvedTime` are nil, and we dereference the pointer later in the code.